### PR TITLE
Fixes: status filters for podman ps and podman pod ps should be inclusive

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>
Fixes: https://github.com/containers/podman/issues/8344#issuecomment-728830411

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
LGTM
@Luap99 PTAL
@containers/podman-maintainers PTAL

File: libpod/runtime_ctr.go [hunk 1]
```diff
@@ -802,13 +802,13 @@ func (r *Runtime) GetContainersWithoutLock(filters ...ContainerFilter) ([]*Conta
...
-			include = include && filter(ctr)
-		}
-
-		if include {
-			ctrsFiltered = append(ctrsFiltered, ctr)
+			// status filters for podman ps and podman pod ps should be inclusive
+			// issue: https://github.com/containers/podman/issues/8344
+			if filter(ctr) {
+				ctrsFiltered = append(ctrsFiltered, ctr)
+				continue
```
Comment (vrothberg) [inline libpod/runtime_ctr.go:810]: Shouldn't this be a `break` to stop iterating over the filters?

Comment (zhangguanzhang) [reply libpod/runtime_ctr.go:810]: got it

**Review by Luap99 — COMMENTED** [review]
This does not address the `podman pod ps` part.

**Comment by Luap99** [comment]
Also doing more testing with docker some filters appear to be inclusive (status) while other are exclusive (label). I need to see if I find something where this is documented.

**Comment by zhangguanzhang** [comment]
> Also doing more testing with docker some filters appear to be inclusive (status) while other are exclusive (label). I need to see if I find something where this is documented.

got it

File: test/e2e/ps_test.go [hunk 2]
```diff
@@ -303,6 +303,22 @@ var _ = Describe("Podman ps", func() {
...
 
+	It("podman status filters should be inclusive", func() {
+		ctr := podmanTest.Podman([]string{"run", "-t", "-i", ALPINE, "ls", "/"})
+		ctr.WaitWithDefaultTimeout()
+		Expect(ctr.ExitCode()).To(Equal(0))
+
+		ctr2 := podmanTest.Podman([]string{"container", "create", ALPINE})
+		ctr2.WaitWithDefaultTimeout()
+		Expect(ctr2.ExitCode()).To(Equal(0))
+
```
Comment (vrothberg) [inline test/e2e/ps_test.go:314]: Can we add another "exited" container? I think we need one item that doesn't match the filter.

**Comment by Luap99** [comment]
The docker logic is quit complex: https://github.com/moby/moby/blob/5c10ea6ae80589c2d56d618df54927e92b0b0136/daemon/list.go#L410

In general filters with the same key are inclusive, e.g. `--filter name=con1 --filter name=con2` returns con1 and con2. The only exception to this seems to be `label` which is exclusive.
Filters with different keys are exclusive, e.g. `--filter name=con1 --filter status=created` will only return con1 if it has the status created.

**Comment by zhangguanzhang** [comment]
I took a look, this logic is too complicated for me🙄。May need someone else to implement this feature

**Comment by Luap99** [comment]
@zhangguanzhang I will.

**Pull request closed without merging**